### PR TITLE
Update existing documentation for publishing 

### PR DESCRIPTION
**Pull request by epananth**

### CODE REVIEW

File: Documentation/CorePackages/Publishing.md [hunk 1]
```diff
@@ -385,3 +381,33 @@ The publishing logs are stored inside an Azure DevOps artifacts container named
 |                     | https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet3/nuget/v3/index.json |
 | dotnet3-transport   | .NET Core 3 non-shipping packages                            |
 |                     | https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet3-transport/nuget/v3/index.json |
+
+
+### How to upgrade to Publishing V3?
```
Comment (riarenas) [inline Documentation/CorePackages/Publishing.md:386]: What is publishing v3?

File: Documentation/CorePackages/Publishing.md [hunk 2]
```diff
@@ -361,7 +357,7 @@ The `DotNetPublishUsingPipelines` is a flag that Arcade SDK uses to determine if
 
 ### Where can I see publishing logs?
 
-The publishing logs are stored inside an Azure DevOps artifacts container named `PostBuildLogs`. Each activated post-build channel/stage will have a subfolder under `PostBuildLogs`. Each job in a publishing channel/stage will have `.binlogs` in the container.
+The publishing logs are stored inside an Azure DevOps artifacts container named `PostBuildLogs`. Publishing stage will have a subfolder under `PostBuildLogs`. Each job in a publishing channel/stage will have `.binlogs` in the container.
```
Comment (riarenas) [inline Documentation/CorePackages/Publishing.md:360]: This isn't accurate in v3, the postbuildLogs only have the validation logs now. You have to go to a different build to find these logs.

File: Documentation/CorePackages/Publishing.md [hunk 3]
```diff
@@ -385,3 +381,33 @@ The publishing logs are stored inside an Azure DevOps artifacts container named
...
+      <PublishingVersion>3</PublishingVersion>
+   </PropertyGroup>
+</Project>
+
+2) Update the publishingInfraVersion in /eng/common/template/post-build/post-build.yml
+
+  publishingInfraVersion: 3
+
+  
+### What is changing in V3?
```
Comment (riarenas) [inline Documentation/CorePackages/Publishing.md:402]: This should probably go before the steps to upgrade to v3

File: Documentation/CorePackages/Publishing.md [hunk 4]
```diff
@@ -385,3 +381,33 @@ The publishing logs are stored inside an Azure DevOps artifacts container named
 |                     | https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet3/nuget/v3/index.json |
 | dotnet3-transport   | .NET Core 3 non-shipping packages                            |
 |                     | https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet3-transport/nuget/v3/index.json |
+
+
+### How to upgrade to Publishing V3?
+
+1) Add PublishingVersion in /eng/Publishing.props
```
Comment (riarenas) [inline Documentation/CorePackages/Publishing.md:388]: This file used to be optional and not all repos had it, I think it's worth calling out that it's now mandatory to have one.

File: Documentation/CorePackages/Publishing.md [hunk 5]
```diff
@@ -385,3 +381,33 @@ The publishing logs are stored inside an Azure DevOps artifacts container named
...
+1) Add PublishingVersion in /eng/Publishing.props 
+
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+   <PropertyGroup>
+      <PublishingVersion>3</PublishingVersion>
+   </PropertyGroup>
+</Project>
+
+2) Update the publishingInfraVersion in /eng/common/template/post-build/post-build.yml
```
Comment (riarenas) [inline Documentation/CorePackages/Publishing.md:397]: This isn't right. You shouldn't change the version here, but rather change it in your repo when post-build.yml is referenced

File: Documentation/CorePackages/Publishing.md [hunk 6]
```diff
@@ -385,3 +381,33 @@ The publishing logs are stored inside an Azure DevOps artifacts container named
...
+
+  publishingInfraVersion: 3
```
Comment (riarenas) [inline Documentation/CorePackages/Publishing.md:399]: It's always helpful to show the whole context, I don't think just adding this line is enough for people to understand what needs to be done, or where this goes. You can also point to the line where arcade is setting this.

File: Documentation/CorePackages/Publishing.md (hunk 3 above)
Comment (riarenas) [reply Documentation/CorePackages/Publishing.md:402]: This section should probably be prefaced by a definition of what V1, v2 and v3 means.

File: Documentation/CorePackages/Publishing.md [hunk 7]
```diff
@@ -385,3 +381,33 @@ The publishing logs are stored inside an Azure DevOps artifacts container named
 |                     | https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet3/nuget/v3/index.json |
 | dotnet3-transport   | .NET Core 3 non-shipping packages                            |
 |                     | https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet3-transport/nuget/v3/index.json |
+
+
+### How to upgrade to Publishing V3?
+
+1) Add PublishingVersion in /eng/Publishing.props 
+
```
Comment (riarenas) [inline Documentation/CorePackages/Publishing.md:389]: It's a good idea to use ``` blocks whenever you are referencing code in a markdown file.

```xml
	<?xml version="1.0" encoding="utf-8"?>
<Project>
   <PropertyGroup>
      <PublishingVersion>3</PublishingVersion>
   </PropertyGroup>
</Project>
```

File: Documentation/CorePackages/Publishing.md [hunk 8]
```diff
@@ -385,3 +381,33 @@ The publishing logs are stored inside an Azure DevOps artifacts container named
...
+</Project>
+
+2) Update the publishingInfraVersion in /eng/common/template/post-build/post-build.yml
+
+  publishingInfraVersion: 3
+
+  
+### What is changing in V3?
+
+Previously the UI was immensely cluttered. Most confusing part was the stages were 'activated' but they don't publish anything. So now we have only one stage for publishing. Publishing stage will have all the information related to errors/warnings/logs.
```
Comment (riarenas) [inline Documentation/CorePackages/Publishing.md:404]: Stick to the same tense. You started with past tense and then moved to present.

Previously the ui was cluttered .... but they didn't publish anything

File: Documentation/Darc.md [hunk 9]
```diff
@@ -879,7 +879,7 @@ task of the "Publish to Build Asset Registry" job in your build. See
 [Locating the BAR build ID for a build](#locating-the-bar-build-id-for-a-build).
 
 ```
-darc add-build-to-channel --id 46856 --channel ".NET Core SDK 3.1.3xx"
+darc add-build-to-channel --id 46856 --channel ".NET Core SDK 3.1.3xx" --publishing-infra-version 3
```
Comment (riarenas) [inline Documentation/Darc.md:882]: Does it matter if a repo is not onboarded to v3 yet?

File: Documentation/Darc.md [hunk 10]
```diff
@@ -1297,16 +1297,18 @@ channel associations (See [add-default-channel](#add-default-channel)).
...
 - Name of the channel you want to assign a build to
 - BAR build id of the build to assign to the specified channel. This can be
   found by looking at the "Publish to Build Asset Registry" leg of an official build.
+- Publishing infra version - 3, for V3 publishing.
 
 **Parameters**
 
 - `--id` - **(Required)**. BAR id of build to assign to channel.
 - `--channel` - **(Required)**. Channel to assign build to.
+- `--publish-infra-version` - **(Required)**. Publishing infrastructure version.
```
Comment (riarenas) [inline Documentation/Darc.md:1311]: There's some missing parameters here:

* `default-channels`
* `signing-validation-parameters`
* `symbol-publishing-parameters`
* `publish-installers-and-checksums`

Comment (riarenas) [reply Documentation/Darc.md:1311]: The Sample needs to be updated, it doesn't reflect the current output

File: Documentation/CorePackages/Publishing.md (hunk 8 above)
Comment (riarenas) [reply Documentation/CorePackages/Publishing.md:404]: [nit] I would stay away from qualifiers such as "immensely", "most" confusing.

**Comment by riarenas** [comment]
I think we need to add sections about the part the build promotion pipeline plays in all this.

**Comment by riarenas** [comment]
The basic onboarding scenario at the top of the document should be updated to reflect onboarding directly to v3. There are two scenarios:

* I'm a new repo and I'm onboarding to publishing
* I'm a repo already publishing with stages

It should be clear how to get into v3 for both this scenarios.

File: Documentation/CorePackages/Publishing.md [hunk 11]
```diff
@@ -385,3 +381,33 @@ The publishing logs are stored inside an Azure DevOps artifacts container named
...
+<Project>
+   <PropertyGroup>
+      <PublishingVersion>3</PublishingVersion>
+   </PropertyGroup>
+</Project>
+
+2) Update the publishingInfraVersion in /eng/common/template/post-build/post-build.yml
+
+  publishingInfraVersion: 3
+
```
Comment (riarenas) [inline Documentation/CorePackages/Publishing.md:400]: What should users expect to see in their builds after making their changes? how do they know it did the right thing?

**Comment by riarenas** [comment]
@epananth should this be closed as there are other PRs for this now?

**Comment by epananth** [comment]
yes closing this.

**Pull request closed without merging**